Tools/ardupilotwaf/mavgen.py

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python
+# encoding: utf-8
+# (c) Siddharth Bharat Purohit, 3DRobotics Inc.
+
+"""
+The **mavgen.py** program is a code generator which creates mavlink header files.
+"""
+
+from waflib import Task
+from waflib.TaskGen import extension
+import os
+
+class mavgen(Task.Task):
+	"""generate mavlink header files"""
+	color   = 'BLUE'
+	run_str = '${MAVGEN} --lang=C --wire-protocol=1.0 --output=generated_src/include/mavlink/v1.0/ ${SRC[0].abspath()}'
+
+@extension('.xml')
+def process_mavgen(self, node):
+	"""
+	Create a mavgen task
+	"""
+	tsk = self.create_task('mavgen', node, [])
+
+def configure(cfg):
+	"""
+	setup environment for mavlink header generator
+	"""
+	env = cfg.env
+	cfg.env.env = dict(os.environ)
+
+	cfg.start_msg('Checking for mavlink submodule')
+	env.MAVLINK_DIR = cfg.srcnode.find_dir('modules/mavlink/').abspath()
+	if not cfg.srcnode.find_dir('modules/mavlink/'):
```

Comment:
The dir can exist in some cases but without any content there. See the gbenchmark tool we added - we look for a specific file inside the submodule that we know should be there. @guludo maybe we should have a helper to check for submodules?